app/league/slack_messenger.py

```diff
@@ -10,24 +13,23 @@ def __init__(self, app=None):
             self.init_app(app)
 
     def init_app(self, app):
+        self.app = app
         self.url = app.config.get('SLACK_WEBHOOK')
         self.channel = app.config.get('SLACK_CHANNEL')
-        self.payload = {'username': 'webhookbot',
-                        'icon_emoji': ':ghost:',
-                        'channel': self.channel}
+        self.username = app.config.get('SLACK_USERNAME') or DEFAULT_USERNAME
+        self.icon_emoji = app.config.get('SLACK_ICON_EMOJI') or DEFAULT_ICON
         app.extensions['messenger'] = self
 
-        def f(msg):
-            """Sends message to Slack."""
-            msg.update(self.payload)
-            try:
-                r = requests.post(self.url, json=self.payload)
-            except requests.exceptions.RequestException as e:
-                pass
-
-        if self.url:
-            self.notify_slack = f
-
     def notify_slack(self, msg):
-        """SLACK_WEBHOOK not initialized."""
-        pass
+        """Sends a simple notification to Slack."""
+        try:
+            payload = {'username': self.username,
+                        'icon_emoji': self.icon_emoji,
+                        'channel': self.channel,
+                        'text': msg}
+            r = requests.post(self.url, json=payload)
+        except AttributeError as e:
+            self.app.logger.warning('Environmental variables for Slack ' +
+                'integration are not configured: {0}'.format(e))
+        except requests.exceptions.RequestException as e:
```

Comment:
Defensively checking and logging is not best practice in general. For this use case, it's true that we don't want the game creation to return badly just because the notification breaks, but that may not always be the case. Better here to catch the exception in `create_game` IMO. Also, we should explicitly prefix the error message indicating the source of the failure (the Slack integration)